tools/check-typo-since

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+#**************************************************************************
+#*                                                                        *
+#*                                 OCaml                                  *
+#*                                                                        *
+#*            Gabriel Scherer, projet Parsifal, INRIA Saclay              *
+#*                                                                        *
+#*   Copyright 2018 Institut National de Recherche en Informatique et     *
+#*     en Automatique.                                                    *
+#*                                                                        *
+#*   All rights reserved.  This file is distributed under the terms of    *
+#*   the GNU Lesser General Public License version 2.1, with the          *
+#*   special exception on linking described in the file LICENSE.          *
+#*                                                                        *
+#**************************************************************************
+
+# Run check-typo, comparing only files that have changed since a particular
+# git state
+
+check_typo_since() {
+  CHECK_TYPO=$(dirname $0)/check-typo
+  for f in $(git diff --name-only $1)
+  do
+    $CHECK_TYPO --check-prune $f 2>/dev/null
+    CHECK_PRUNE=$?
+    if [[ -e $f && $CHECK_PRUNE > 0 ]]
```

Comment:
I'm fairly sure `[[` is a bashism - though I'd change the shebang to be `#! /usr/bin/env bash` rather than remove the bashism.

Reply:
Thanks for the catch! I'll use the suggested shebang.